app/controllers/users/emails_controller.rb

```diff
@@ -94,6 +99,7 @@ def handle_successful_delete
     end
 
     def process_successful_creation
+      session.delete(:in_select_email_flow)
```

Comment:
This'll probably make it so that the `in_select_email_flow` detail gets dropped if someone clicks "Resend" or "use a different email address" from the success confirmation screen.